importApp/src/main/java/com/commercetools/dataimport/joyrideavailability/AvailabilityJobConfiguration.java

```diff
@@ -0,0 +1,132 @@
+import org.springframework.batch.core.step.builder.StepBuilder;
+import org.springframework.batch.item.ItemProcessor;
+import org.springframework.batch.item.ItemReader;
+import org.springframework.batch.item.ItemStreamReader;
+import org.springframework.batch.item.ItemWriter;
+import org.springframework.batch.item.file.FlatFileItemReader;
+import org.springframework.batch.item.file.mapping.BeanWrapperFieldSetMapper;
+import org.springframework.batch.item.file.mapping.DefaultLineMapper;
+import org.springframework.batch.item.file.transform.DelimitedLineTokenizer;
+import org.springframework.batch.item.support.ListItemReader;
+import org.springframework.batch.item.support.SynchronizedItemStreamReader;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.core.io.Resource;
+import org.springframework.core.task.SimpleAsyncTaskExecutor;
+import org.springframework.core.task.TaskExecutor;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static io.sphere.sdk.client.SphereClientUtils.blockingWait;
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.toList;
+
+@Component
+@Lazy
+public class AvailabilityJobConfiguration extends DefaultCommercetoolsJobConfiguration {
+    private static final Logger logger = LoggerFactory.getLogger(AvailabilityJobConfiguration.class);
+
+    @Bean
+    public Job productsSuggestionsCopyJob(final Step channelImportStep, final Step customTypeImportStep) {
+        return jobBuilderFactory.get("importJoyrideAvailabilityJob")
+                  .start(customTypeImportStep)
+                  .next(channelImportStep)
+                .build();
+    }
+
+    @Bean
+    public Step channelImportStep(final BlockingSphereClient sphereClient,
+                                  final ItemReader<ChannelDraftDsl> channelsDraftReader,
+                                  final ItemWriter<ChannelDraftDsl> channelsDraftItemWriter) {
+        final StepBuilder stepBuilder = stepBuilderFactory.get("channelImportStep");
+        return stepBuilder
+                .<ChannelDraftDsl, ChannelDraftDsl>chunk(1)
+                .reader(channelsDraftReader)
+                .writer(channelsDraftWriter(sphereClient))
+                .build();
+    }
+
+    @Bean
+    @StepScope
+    private ItemReader<ChannelDraftDsl> channelsDraftReader(@Value("#{jobParameters['resource']}") final Resource channelsJsonResource) throws IOException {
+        logger.info("URL_Channels: " + channelsJsonResource);
+        final ObjectReader reader = SphereJsonUtils.newObjectMapper().readerFor(new TypeReference<List<ChannelDraftDsl>>() {
+        });
+        final InputStream inputStream = channelsJsonResource.getInputStream();
```

Comment:
Jackson might close it for you: `https://github.com/FasterXML/jackson-databind/issues/683`

never mind